crates/ruma-state-res/src/event_auth.rs

```diff
@@ -223,7 +223,20 @@ pub fn auth_check<E: Event>(
         return Ok(false);
     }
 
-    // [synapse] checks for federation here
+    // If the create event content has the field m.federate set to false and the sender domain of
+    // the event does not match the sender domain of the create event, reject.
+    #[derive(Deserialize)]
+    struct RoomCreateContentFederate {
+        federate: bool,
```

Comment:
This seems to be missing a `serde(rename = "m.federate")`, and shouldn't it be optional / `serde(default)` too?